### DISCUSSION TITLE
How to use Heat from the command line

### DISCUSSION THREAD

**Discussion by jmairboeck** [wixtoolset/issues#7479] [Questions] [answered]
How can Heat be used from the command line in Wix 4?

It seems to be only available as a NuGet package `WixToolset.Heat`. After it has been added to a project, it is available in `%USERPROFILE%\.nuget\packages\wixtoolset.heat\4.0.0`, but how do I get this without creating a project first?

Could Heat be provided as a dotnet tool, similar to `wix.exe`?

**Comment by barnson** [comment]
It's included in the WixAdditionalTools.exe bundle. https://github.com/wixtoolset/wix/releases/tag/v4.0.0

**Reply by jmairboeck** [reply]
> It's included in the WixAdditionalTools.exe bundle. https://github.com/wixtoolset/wix/releases/tag/v4.0.0

That contains only thmviewer.exe in the bin folder.

**Reply by barnson** [reply]
Huh. Sorry, I guess my memory's shot or plans changed.

**Comment by jcasale** [comment]
I was just about to post the same question as I stumbled over the same revelation when upgrading a few projects.

**Comment by computerquip-work** [comment]
This is still a problem. I don't use the MSBuild-style projects and have a separate script that compiles and links the MSI files together. The Heat tool can't be installed via `dotnet tool install` and trying to manually add it as an extension will show up as "damaged" for probably good reasons.

What I've had to do instead is install the tool manually via `nuget` or `Install-Package` then manually add the directory with heat.exe to PATH. Not the most elegant I'd have to say.

**Reply by llnagy** [reply]
"What I've had to do instead is install the tool manually via nuget or Install-Package then manually add the directory with heat.exe to PATH. Not the most elegant I'd have to say."

Were you able to install a version 4 heat.exe? If so, what command did you use? I would also like to use heat on the command line to convert an old project. I've tried using the HarvestDirectory in the MSBuild file and have not been able to get it working.

**Reply by jcasale** [reply]
There are many ways, but ultimately you need to get the package downloaded and extracted, for example:

```PowerShell
PS C:\> nuget install WixToolset.Heat -OutputDirectory c:\tmp
PS C:\> Get-ChildItem C:\tmp\WixToolset.Heat* -Include heat.exe -Recurse |select FullName

FullName
--------
C:\tmp\WixToolset.Heat.4.0.1\tools\net472\x64\heat.exe
C:\tmp\WixToolset.Heat.4.0.1\tools\net472\x86\heat.exe
```

**Reply by computerquip** [reply, marked as answer]
For what it's worth, I caved and ended up converting to an MSBuild project so it ended up not being a problem for me directly anymore. I've also been told (per https://github.com/wixtoolset/issues/issues/6838) that Heat in its current incarnation will be obsoleted at some future version.
